### ISSUE TITLE
unable to scroll listview till the end if panel is anchored

### ISSUE THREAD

**Issue by jihuncha** [umano/AndroidSlidingUpPanel#967]
i downloaded demo file and make some change to scroll my listview at anchored point. but it didn't scroll till the end.
 i found 'Scrollable Sliding Views' and add my xml umanoScrollableView="@+id/list"
but it is not helpful. how can i solve this problem??

![Screenshot_20190608-205308](https://user-images.githubusercontent.com/38685544/59146966-e0230c00-8a2f-11e9-929a-51aeb0acd020.png)

plz help me!!

**Comment by user2pc6** [comment]
That is because u added the anchor point, it would be a point where the layout stops. is it ok if u attach ur xml layout for me  to take look?

**Comment by peter-yeh** [comment]
That is because u added the anchor point, it would be a point where the layout stops. is it ok if u attach ur xml layout for me to take look?

**Comment by peter-yeh** [comment]
`<com.sothree.slidinguppanel.SlidingUpPanelLayout

    
    <FrameLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:orientation="vertical">
    </FrameLayout>

    
    <LinearLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:background="#ffffff"
        android:orientation="vertical"
        android:clickable="true"
        android:focusable="false"
        android:id="@+id/dragView">

        <LinearLayout
            android:layout_width="match_parent"
            android:layout_height="68dp"
            android:orientation="horizontal">

            <TextView
                android:id="@+id/name"
                android:layout_width="0dp"
                android:layout_height="match_parent"
                android:layout_weight="1"
                android:textSize="14sp"
                android:gravity="center_vertical"
                android:paddingLeft="10dp"/>

            <Button
                android:id="@+id/follow"
                android:layout_width="wrap_content"
                android:layout_height="match_parent"
                android:textSize="14sp"
                android:gravity="center_vertical|right"
                android:paddingRight="10dp"
                android:paddingLeft="10dp"/>

        </LinearLayout>

        <ListView
            android:id="@+id/list"
            android:layout_width="match_parent"
            android:layout_height="0dp"
            android:layout_weight="1">
        </ListView>

        
            
            
            
            
            
      
        
    </LinearLayout>
</com.sothree.slidinguppanel.SlidingUpPanelLayout>`


This is the demo xml provided by umano, I truncated some codes. as  u can see  the     is a linear layout, linenar layout do not contain scroll view. that is why it does not scroll. Adding a scroll view would make it scroll

**Comment by jihuncha** [comment]
> `<com.sothree.slidinguppanel.SlidingUpPanelLayout
> 
> ```
> 
> <FrameLayout
>     android:layout_width="match_parent"
>     android:layout_height="match_parent"
>     android:orientation="vertical">
> </FrameLayout>
> 
> 
> <LinearLayout
>     android:layout_width="match_parent"
>     android:layout_height="match_parent"
>     android:background="#ffffff"
>     android:orientation="vertical"
>     android:clickable="true"
>     android:focusable="false"
>     android:id="@+id/dragView">
> 
>     <LinearLayout
>         android:layout_width="match_parent"
>         android:layout_height="68dp"
>         android:orientation="horizontal">
> 
>         <TextView
>             android:id="@+id/name"
>             android:layout_width="0dp"
>             android:layout_height="match_parent"
>             android:layout_weight="1"
>             android:textSize="14sp"
>             android:gravity="center_vertical"
>             android:paddingLeft="10dp"/>
> 
>         <Button
>             android:id="@+id/follow"
>             android:layout_width="wrap_content"
>             android:layout_height="match_parent"
>             android:textSize="14sp"
>             android:gravity="center_vertical|right"
>             android:paddingRight="10dp"
>             android:paddingLeft="10dp"/>
> 
>     </LinearLayout>
> 
>     <ListView
>         android:id="@+id/list"
>         android:layout_width="match_parent"
>         android:layout_height="0dp"
>         android:layout_weight="1">
>     </ListView>
> 
>     
>         
>         
>         
>         
>         
>   
>     
> </LinearLayout>
> ```
> 
> </com.sothree.slidinguppanel.SlidingUpPanelLayout>`
> 
> This is the demo xml provided by umano, I truncated some codes. as u can see the is a linear layout, linenar layout do not contain scroll view. that is why it does not scroll. Adding a scroll view would make it scroll


thanks for answering my question. but i dont understand what you talking about. you mean i have to add scrollview?? in linearlayout??

here is my xml code. but i think there is no change between umano code. 

my purpose is that my listview have to scroll all contents at ANCHORED point but it didnt work.

`<com.sothree.slidinguppanel.SlidingUpPanelLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    xmlns:sothree="http://schemas.android.com/apk/res-auto"
    android:id="@+id/sliding_layout"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:gravity="bottom"
    sothree:umanoPanelHeight="68dp"
    sothree:umanoShadowHeight="4dp"
    sothree:umanoParallaxOffset="100dp"
    sothree:umanoDragView="@+id/dragView"
    sothree:umanoOverlay="true"
    sothree:umanoScrollableView="@+id/list">

    
    <FrameLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:orientation="vertical">
        <android.support.v7.widget.Toolbar
            xmlns:sothree="http://schemas.android.com/apk/res-auto"
            xmlns:android="http://schemas.android.com/apk/res/android"
            android:id="@+id/main_toolbar"
            android:layout_height="?attr/actionBarSize"
            android:background="?attr/colorPrimary"
            sothree:theme="@style/ActionBar"
            android:layout_width="match_parent"/>
        <TextView
            android:id="@+id/main"
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            android:layout_marginTop="?attr/actionBarSize"
            android:gravity="center"
            android:text="Main Content"
            android:clickable="true"
            android:focusable="false"
            android:focusableInTouchMode="true"
            android:textSize="16sp" />
    </FrameLayout>

    
    <LinearLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:background="#ffffff"
        android:orientation="vertical"
        android:clickable="true"
        android:focusable="false"
        android:id="@+id/dragView">

        <LinearLayout
            android:layout_width="match_parent"
            android:layout_height="68dp"
            android:orientation="horizontal">

            <TextView
                android:id="@+id/name"
                android:layout_width="0dp"
                android:layout_height="match_parent"
                android:layout_weight="1"
                android:textSize="14sp"
                android:gravity="center_vertical"
                android:paddingLeft="10dp"/>

            <Button
                android:id="@+id/follow"
                android:layout_width="wrap_content"
                android:layout_height="match_parent"
                android:textSize="14sp"
                android:gravity="center_vertical|right"
                android:paddingRight="10dp"
                android:paddingLeft="10dp"/>

        </LinearLayout>

        <ListView
            android:id="@+id/list"
            android:layout_width="match_parent"
            android:layout_height="0dp"
            android:layout_weight="1">
        </ListView>


        
            
            
            
            
            
            
                
                
                
        
    </LinearLayout>
</com.sothree.slidinguppanel.SlidingUpPanelLayout>`

**Comment by yuchun98** [comment]
yes, in order for the list view to scroll at anchor point, u need to add a view that enables scrolling. this is only achievable by adding the scrollview to ur listview

**Comment by yuchun98** [comment]
This would be an example 
`<ScrollView
    android:layout_width="wrap_content"
    android:layout_height="wrap_content">
    <ListView
        android:id="@+id/list"
        android:layout_width="match_parent"
        android:layout_height="0dp"
        android:layout_weight="1">
    </ListView>
    
</ScrollView>`

**Comment by yuchun98** [comment]
This is an example

<ScrollView
    android:layout_width="wrap_content"
    android:layout_height="wrap_content">
    <ListView
        android:id="@+id/list"
        android:layout_width="match_parent"
        android:layout_height="0dp"
        android:layout_weight="1">
    </ListView>
    
</ScrollView>

**Comment by yuchun98** [comment]
<ScrollView
    android:layout_width="wrap_content"
    android:layout_height="wrap_content">
    <ListView
        android:id="@+id/list"
        android:layout_width="match_parent"
        android:layout_height="0dp"
        android:layout_weight="1">
    </ListView>
    
</ScrollView>

**Comment by peter-yeh** [comment]
I don't get it, u did not add anchor point anywhere in ur code. originally it was already working for me

**Comment by cloudchenc** [comment]
hi, I have the same issue! I use the link to fix that!
https://github.com/umano/AndroidSlidingUpPanel/issues/873#issuecomment-463889055
